### PR TITLE
MGMT-1814 - Add pull secret validation on create cluster

### PR DESCRIPTION
**Pull request by rollandf**
In case that Auth is enabled:

- Validate that the pull secret has a "cloud.openshift.com" section
- Validate that the pull secret user token user is the same as the user that initiated the Cluster creation

### CODE REVIEW

File: cmd/main.go [hunk 1]
```diff
@@ -189,7 +189,7 @@ func main() {
 		Options.BMConfig.S3EndpointURL = newUrl
 	}
 
-	bm := bminventory.NewBareMetalInventory(db, log.WithField("pkg", "Inventory"), hostApi, clusterApi, Options.BMConfig, generator, eventsHandler, objectHandler, metricsManager)
+	bm := bminventory.NewBareMetalInventory(db, log.WithField("pkg", "Inventory"), hostApi, clusterApi, Options.BMConfig, generator, eventsHandler, objectHandler, metricsManager, *authHandler)
```
Comment (filanov) [inline cmd/main.go:192]: please break the line

File: internal/cluster/validations/validations.go [hunk 2]
```diff
@@ -72,11 +76,25 @@ const (
...
+func ValidatePullSecret(secret string, username string, authHandler auth.AuthHandler) error {
+	creds, err := ParsePullSecret(secret)
 	if err != nil {
 		return err
 	}
+
+	if authHandler.EnableAuth {
+		r, ok := creds["cloud.openshift.com"]
+		if !ok {
+			return fmt.Errorf("Pull secret does not contain auth for cloud.openshift.com")
```
Comment (filanov) [inline internal/cluster/validations/validations.go:88]: please use `github.com/pkg/errors` for errors.

File: internal/cluster/validations/validations.go [hunk 3]
```diff
@@ -72,11 +76,25 @@ const (
...
+		}
+		user, err := authHandler.AuthAgentAuth(r.AuthRaw)
+		if err != nil {
+			return fmt.Errorf("Failed to authenticate Pull Secret Token")
+		}
+		if (user.(*ocm.AuthPayload)).Username != username {
```
Comment (filanov) [inline internal/cluster/validations/validations.go:94]: We don't intent to support creation of cluster for other user?

Comment (rollandf) [reply internal/cluster/validations/validations.go:94]: Not for now. The Authz part will not allow to see other user clusters.
@avishayt please correct me if I am wrong.

See additional discussion in JIRA MGMT-1814

File: internal/cluster/validations/validations.go (hunk 2 above)
Comment (rollandf) [reply internal/cluster/validations/validations.go]: Done

File: cmd/main.go (hunk 1 above)
Comment (rollandf) [reply cmd/main.go]: Done

**Review by filanov — APPROVED** [review]

**Review by filanov — APPROVED** [review]

**Review by filanov — APPROVED** [review]

**Pull request merged**